documentation/proposals/Proposal - Generic Math.md

```diff
@@ -0,0 +1,1182 @@
+                           IVector2<Vector2I<TScalar>, TScalar>
+                           where TScalar : IBinaryInteger<TScalar>
+{
+    ///<summary>
+    ///The X component of the Vector
+    ///</summary>
+    [DataMember]
+    public TScalar X { get; }
+
+    ///<summary>
+    ///The Y component of the Vector
+    ///</summary>
+    [DataMember]
+    public TScalar Y { get; }
+
+    public static Vector2I<TScalar> UnitX { get; }
+    public static Vector2I<TScalar> UnitY { get; }
+
+    public static Vector2I<TScalar> Zero { get; }
+    public static Vector2I<TScalar> One { get; }
+
+    public TScalar this[int index] { get; }
+
+    public Vector2I(TScalar x, TScalar y);
+
+    public static Vector2I<TScalar> operator +(Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> operator -(Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> operator *(Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> operator /(Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> operator %(Vector2I<TScalar> left, Vector2I<TScalar> right);
+
+    public static Vector2I<TScalar> operator *(Vector2I<TScalar> left, TScalar right);
+    public static Vector2I<TScalar> operator /(Vector2I<TScalar> left, TScalar right);
+    public static Vector2I<TScalar> operator %(Vector2I<TScalar> left, TScalar right);
+
+    public static Vector2I<TScalar> operator *(TScalar left, Vector2I<TScalar> right);
+
+    public static Vector2I<TScalar> operator +(Vector2I<TScalar> value);
+    public static Vector2I<TScalar> operator -(Vector2I<TScalar> value);
+
+    public static Vector2I<TScalar> operator &(Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> operator |(Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> operator ^(Vector2I<TScalar> left, Vector2I<TScalar> right);
+
+    public static Vector2I<TScalar> operator ~(Vector2I<TScalar> value);
+
+    public TScalar LengthSquared { get; }
+
+    public Vector2I<TScalar> Dot(Vector2I<TScalar> other);
+    public TScalar DistanceSquared(Vector2I<TScalar> other);
+    public Vector2I<TScalar> Abs();
+    public void CopyTo(TScalar[] array);
+    public void CopyTo(TScalar[] array, int index);
+
+    public static Vector2I<TScalar> Dot (Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> DistanceSquared (Vector2I<TScalar> left, Vector2I<TScalar> right);
+    public static Vector2I<TScalar> Clamp (Vector2I<TScalar> value, Vector2I<TScalar> min, Vector2I<TScalar> max);
+    public static Vector2I<TScalar> Min (Vector2I<TScalar> value1, Vector2I<TScalar> value2);
+    public static Vector2I<TScalar> Max (Vector2I<TScalar> value1, Vector2I<TScalar> value2);
+    public static Vector2I<TScalar> Abs (Vector2I<TScalar> value);
```

Comment:
These methods will be provided on a static class. They will have explicit implementations of the IVector static abstract methods that simply delegate to the static class.

Reply:
This is just to save space. I combined the static classes with their relevant type. This does not have to match the implementation. It is just to get across what functions we expect to be available.